### PR TITLE
Teleport was constantly asking for username+HOTP+password

### PR DESCRIPTION
**Pull request by kontsevoy**
The problem was in `time.Unix()` instead of `time.Unixnano()`
I'm not even sure how Unix() got in there...

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -94,9 +94,6 @@ type Config struct {
 	// or use any local certs, and not use interactive logins
 	SkipLocalAuth bool
 
-	// AuthMethods if passed will be used to login into cluster
-	AuthMethods []ssh.AuthMethod
-
```
Comment (kontsevoy) [inline lib/client/api.go]: This field was never used.

Comment (klizhentas) [inline lib/client/api.go]: Can you bring it back? I use it in the library mode, when passing external authenticators (not local auth)

**Comment by klizhentas** [comment]
lgtm except that I need this `AuthMethods` thing to exist

**Comment by kontsevoy** [comment]
Mm... this can be greatly simplified. Why do you need a separate boolean flag for "skipLocalAuth" then? If the purpose of this to supply your own auth methods, why the need for two parameters?

**Comment by kontsevoy** [comment]
Ok, I brought back the original Config.AuthMethods, but removed tc.authMethods to avoid duplication.

**Comment by kontsevoy** [comment]
retest this

**Comment by kontsevoy** [comment]
retest this please

**Comment by klizhentas** [comment]
SkipLocalAuth tells client not to resort to local auth method if it failed everything else here:

https://github.com/gravitational/teleport/blob/master/lib/client/api.go#L704

Lgtm

**Pull request merged**